File: spark-parctise/3-spark-rdd/src/main/scala/com/westar/spark/rdd/wordcount/WordCount.scala
package com.westar.spark.rdd.wordcount

import org.apache.spark.{SparkConf, SparkContext}

object WordCount {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf()
    conf.setAppName("WordCount")

    val sparkContext = new SparkContext(conf)

    val textFileRdd = sparkContext.textFile("hdfs://hadoop0:9000/user/hadoop/word.txt")

    val wordRDD = textFileRdd.flatMap(line => line.split(" "))

    val pairWordRDD = wordRDD.map(word => (word, 1))

    val wordCountRDD = pairWordRDD.reduceByKey( _ + _)

    wordCountRDD.saveAsTextFile("hdfs://hadoop0:9000/user/hadoop/wordcount")


  }

}

File: spark-parctise/6.2-spark-rdd-python/zzqpyspark/wordcount.py
from pyspark import SparkContext, SparkConf

import os
import shutil

if __name__ == '__main__':
    conf = SparkConf().setAppName("appName").setMaster("local")
    sc = SparkContext(conf=conf)

    sourceDataRDD = sc.textFile("E:/IDEAProject/hadoopProject/spark-parctise/9-spark-dataset/src/main/resources/环境安装.txt")

    wordsRDD = sourceDataRDD.flatMap(lambda line: line.split())

    keyValueWordsRDD = sourceDataRDD.map(lambda word: (word,1))

    wordCountRDD =  keyValueWordsRDD.reduceByKey(lambda a, b: a + b)

    outputPath = "E:/IDEAProject/hadoopProject/spark-parctise/9-spark-dataset/src/main/resources/wordcountPyth"
    if os.path.exists(outputPath):
        shutil.rmtree(outputPath)

    wordCountRDD.saveAsTextFile(outputPath)

    print(wordCountRDD.collect())